eng/config/BannedSymbols.txt

```diff
@@ -7,6 +7,7 @@ T:Microsoft.CodeAnalysis.Recommendations;RecommendationOptions; Use Recommendati
 T:Microsoft.CodeAnalysis.Formatting.FormattingOptions; Use FormattingOptions2 instead
 T:Microsoft.CodeAnalysis.Options.PerLanguageOption`1; Use PerLanguageOption2<T> instead
 T:Microsoft.CodeAnalysis.Options.Option`1; Use 'Microsoft.CodeAnalysis.Options.Option2' instead
+T:Microsoft.CodeAnalysis.SyntaxWalker; Use SyntaxNode.DescendantNodes methods instead.
```

Comment:
What we are going to observe when the API is used?

Reply:
The build will fail if there are references to this class.

Reply:
> The build will fail if there are references to this class.

Will the failure clearly explain the reason?

Reply:
Hopefully it's clear. Below is an image from the issue expressed inside VS (the command line build succeeds on my machine, but it will fail on ci -- I know that because I originally didn't change the comment in SyntaxWalkerDepth and it was flagged on the ci)

![image](https://github.com/dotnet/roslyn/assets/6785178/e3de0598-55fd-40f3-bda6-757ee922bf74)